submission_offsets.py

```diff
@@ -0,0 +1,68 @@
+from flask.cli import FlaskGroup
+import click
+from collections import defaultdict
+
+import db
+import webserver
+
+from sqlalchemy import text
+
+cli = FlaskGroup(add_default_commands=False, create_app=webserver.create_app_flaskgroup)
+
+@cli.command(name='add-offsets')
+@click.option("--limit", "-l", default=10000)
+def add_offsets(limit):
+	"""Update lowlevel submission offsets with a specified limit."""
+	incremental_add_offset(limit)
+
+def incremental_add_offset(limit):
+	with db.engine.connect() as connection:
+		# Find the next batch of items to update
+		batch_query = text("""
+			SELECT id, gid
+			  FROM lowlevel
+			 WHERE submission_offset IS NULL
+			 LIMIT :limit
+		""")
+		batch_result = connection.execute(batch_query, { "limit": limit })
+
+		# Find max existing offsets
+		offset_query = text("""
```

Comment:
if we keep the max_offsets dictionary, we only need to do this query once when the script starts up, instead of for each 10,000 items